internal/services/cdn/azuresdkhacks/cdn_frontdoor_origin_group.go

```diff
@@ -0,0 +1,132 @@
+}
+
+// OriginGroupUpdateParameters origin group properties needed for origin group creation or update.
+type AFDOriginGroupUpdateParameters struct {
+	*AFDOriginGroupUpdatePropertiesParameters `json:"properties,omitempty"`
+}
+
+// MarshalJSON is the custom marshaler for OriginGroupUpdateParameters.
+func (ogup AFDOriginGroupUpdateParameters) MarshalJSON() ([]byte, error) {
+	objectMap := make(map[string]interface{})
+	if ogup.AFDOriginGroupUpdatePropertiesParameters != nil {
+		objectMap["properties"] = ogup.AFDOriginGroupUpdatePropertiesParameters
+	}
+	return json.Marshal(objectMap)
+}
+
+// UnmarshalJSON is the custom unmarshaler for AFDOriginGroupUpdateParameters struct.
+func (ogup *AFDOriginGroupUpdateParameters) UnmarshalJSON(body []byte) error {
+	var m map[string]*json.RawMessage
+	err := json.Unmarshal(body, &m)
+	if err != nil {
+		return err
+	}
+	for k, v := range m {
+		switch k { // nolint gocritic
+		case "properties":
+			if v != nil {
+				var afdOriginGroupUpdatePropertiesParameters AFDOriginGroupUpdatePropertiesParameters
+				err = json.Unmarshal(*v, &afdOriginGroupUpdatePropertiesParameters)
+				if err != nil {
+					return err
+				}
+				ogup.AFDOriginGroupUpdatePropertiesParameters = &afdOriginGroupUpdatePropertiesParameters
+			}
+		}
+	}
+
+	return nil
+}
+
+// AFDOriginGroupUpdatePropertiesParameters the JSON object that contains the properties of the origin group.
+type AFDOriginGroupUpdatePropertiesParameters struct {
+	// ProfileName - READ-ONLY; The name of the profile which holds the origin group.
+	ProfileName *string `json:"profileName,omitempty"`
+	// LoadBalancingSettings - Load balancing settings for a backend pool
+	LoadBalancingSettings *cdn.LoadBalancingSettingsParameters `json:"loadBalancingSettings,omitempty"`
+	// HealthProbeSettings - Health probe settings to the origin that is used to determine the health of the origin.
+	HealthProbeSettings *cdn.HealthProbeParameters `json:"healthProbeSettings,omitempty"`
+	// TrafficRestorationTimeToHealedOrNewEndpointsInMinutes - Time in minutes to shift the traffic to the endpoint gradually when an unhealthy endpoint comes healthy or a new endpoint is added. Default is 10 mins. This property is currently not supported.
+	TrafficRestorationTimeToHealedOrNewEndpointsInMinutes *int32 `json:"trafficRestorationTimeToHealedOrNewEndpointsInMinutes,omitempty"`
+	// SessionAffinityState - Whether to allow session affinity on this host. Valid options are 'Enabled' or 'Disabled'. Possible values include: 'EnabledStateEnabled', 'EnabledStateDisabled'
+	SessionAffinityState cdn.EnabledState `json:"sessionAffinityState,omitempty"`
+}
+
+// MarshalJSON is the custom marshaler for AFDOriginGroupUpdatePropertiesParameters.
+func (ogupp AFDOriginGroupUpdatePropertiesParameters) MarshalJSON() ([]byte, error) {
+	objectMap := make(map[string]interface{})
+	if ogupp.LoadBalancingSettings != nil {
+		objectMap["loadBalancingSettings"] = ogupp.LoadBalancingSettings
+	}
```

Comment:
presumably this is also the case for loadBalancingSettings here too?

Reply:
Good question, but I do not believe it will be needed for the loadBalancingSettings since it is a required field and has default values for the attributes within the code block. On a side note, it appears that the loadBalincingSettings are mostly used by the Health Probes to evaluate the quickest route to the nearest backend pool. So going out on a limb here, if you disable the health probes the loadBalincingSettings are ignored?